### PR TITLE
feat: allow use of async proxy methods

### PR DESCRIPTION
**Pull request by sla100**
Hello Markus.
I've been testing your amazing project for a few days now. While using the proxy, I encountered the problem of having to implement the proxy with synchronous methods.

I tested locally in my async function support solution and it seems to work fine. Please see if this could be added to the project.
Regards.

### CODE REVIEW

**Review by MarkusJx — CHANGES_REQUESTED** [review]

**Review by MarkusJx — APPROVED** [review]

**Comment by sla100** [comment]
Thank you. I will be waiting for the next release.

**Comment by MarkusJx** [comment]
But now the changes you've made won't be in the next release as you've closed this MR before it has been merged. I'm confused.

**Comment by sla100** [comment]
> But now the changes you've made won't be in the next release as you've closed this MR before it has been merged. I'm confused.

My bad. Sorry!

**Review by MarkusJx — APPROVED** [review]

**Pull request merged**
